distribution/test/itests/test-itests-ddf-core/src/test/java/ddf/test/itests/core/DdfCoreIT.java

```diff
@@ -0,0 +1,471 @@
+
+import static com.jayway.restassured.RestAssured.get;
+import static com.jayway.restassured.RestAssured.given;
+import static com.jayway.restassured.RestAssured.when;
+import static org.awaitility.Awaitility.await;
+import static org.codice.ddf.itests.common.AbstractIntegrationTest.DynamicUrl.INSECURE_ROOT;
+import static org.codice.ddf.itests.common.catalog.CatalogTestCommons.ingest;
+import static org.codice.ddf.itests.common.csw.CswTestCommons.CSW_FEDERATED_SOURCE_FACTORY_PID;
+import static org.codice.ddf.itests.common.csw.CswTestCommons.GMD_CSW_FEDERATED_SOURCE_FACTORY_PID;
+import static org.codice.ddf.itests.common.csw.CswTestCommons.getCswInsertRequest;
+import static org.codice.ddf.itests.common.csw.CswTestCommons.getCswQuery;
+import static org.codice.ddf.itests.common.csw.CswTestCommons.getCswSourceProperties;
+import static org.codice.ddf.itests.common.csw.CswTestCommons.getMetacardIdFromCswInsertResponse;
+import static org.codice.ddf.itests.common.opensearch.OpenSearchTestCommons.OPENSEARCH_FACTORY_PID;
+import static org.codice.ddf.itests.common.opensearch.OpenSearchTestCommons.getOpenSearch;
+import static org.codice.ddf.itests.common.opensearch.OpenSearchTestCommons.getOpenSearchSourceProperties;
+import static org.hamcrest.CoreMatchers.containsString;
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.CoreMatchers.startsWith;
+import static org.hamcrest.xml.HasXPath.hasXPath;
+
+import com.google.common.collect.ImmutableMap;
+import com.jayway.restassured.http.ContentType;
+import com.jayway.restassured.response.Response;
+import com.xebialabs.restito.server.StubServer;
+import com.xebialabs.restito.server.secure.SecureStubServer;
+import ddf.catalog.data.Metacard;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+import javax.ws.rs.core.HttpHeaders;
+import javax.ws.rs.core.MediaType;
+import javax.xml.xpath.XPathExpressionException;
+import org.apache.commons.io.FileUtils;
+import org.codice.ddf.itests.common.AbstractIntegrationTest;
+import org.codice.ddf.itests.common.config.UrlResourceReaderConfigurator;
+import org.codice.ddf.itests.common.csw.mock.FederatedCswMockServer;
+import org.codice.ddf.test.common.annotations.AfterExam;
+import org.codice.ddf.test.common.annotations.BeforeExam;
+import org.hamcrest.Matchers;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.ops4j.pax.exam.junit.PaxExam;
+import org.ops4j.pax.exam.spi.reactors.ExamReactorStrategy;
+import org.ops4j.pax.exam.spi.reactors.PerSuite;
+
+@RunWith(PaxExam.class)
+@ExamReactorStrategy(PerSuite.class)
+public class DdfCoreIT extends AbstractIntegrationTest {
```

Comment:
❔ Should we comment on why these are considered Core tests? What are the criteria or how we determine what is a `core` test in the future? Not sure if belongs here, but seems like it would be great for anybody who gets into this file.

Reply:
Ideally we would try to only have a set of a dozen end to end tests going forward.  It would be good to have a comment that for something to be added, something needs to be brought down to a lower level.  The only things tested here should be those things that can only be tested end to end and are critical for a functional system.  Everything else should be added at a lower level of the test pyramid.